internal/fullnode/stuck_detection.go

```diff
@@ -0,0 +1,121 @@
+package fullnode
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	cosmosv1 "github.com/strangelove-ventures/cosmos-operator/api/v1"
+	"github.com/strangelove-ventures/cosmos-operator/internal/kube"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type StuckPodDetection struct {
+	available      func(pods []*corev1.Pod, minReady time.Duration, now time.Time) []*corev1.Pod
+	collector      StatusCollector
+	computeRollout func(maxUnavail *intstr.IntOrString, desired, ready int) int
+}
+
+func NewStuckDetection(collector StatusCollector) StuckPodDetection {
+	return StuckPodDetection{
+		available:      kube.AvailablePods,
+		collector:      collector,
+		computeRollout: kube.ComputeRollout,
+	}
+}
+
+// StuckPods returns pods that are stuck on a block height due to a cometbft issue that manifests on sentries using horcrux.
+func (d StuckPodDetection) StuckPods(ctx context.Context, crd *cosmosv1.CosmosFullNode) []*corev1.Pod {
+	pods := d.collector.Collect(ctx, client.ObjectKeyFromObject(crd)).Synced().Pods()
+
+	for i, pod := range pods {
+		config, err := rest.InClusterConfig()
+		if err != nil {
+			panic(err.Error())
+		}
+
+		clientset, err := kubernetes.NewForConfig(config)
+		if err != nil {
+			panic(err.Error())
+		}
+
+		receivedString := getPodLogsLastLine(clientset, pod)
+		fmt.Println(receivedString)
+		podIsStuck := isPodStuck(receivedString)
+
+		//MORE TODO HERE
+		if podIsStuck {
+			pods = removeElement(pods, i)
+		}
+	}
+	return pods
+}
+
+func isPodStuck(receivedString string) bool {
+	if strings.Contains(receivedString, "SignerListener: Connected") {
+		timeInLog, err := extractTimeFromLog(receivedString)
+		if err != nil {
+			fmt.Println("Error parsing time from log:", err)
+			return true
```

Comment:
We may not want to return `true` here, if the time parse fails this will return true from isPodStuck.

Can we change this function to return `(bool, error)` so we can track the errors better?